sentry_sdk/profiler.py

```diff
@@ -778,27 +778,26 @@ def __exit__(self, ty, value, tb):
         # type: (Optional[Any], Optional[Any], Optional[Any]) -> None
         self.teardown()
 
+    @abstractmethod
     def setup(self):
         # type: () -> None
-        raise NotImplementedError
+        pass
 
+    @abstractmethod
     def teardown(self):
         # type: () -> None
-        raise NotImplementedError
+        pass
 
+    @abstractmethod
     def ensure_running(self):
         # type: () -> None
-        raise NotImplementedError
+        pass
```

Comment:
I'd not turn `ensure_running` into an `abstractmethod`. `setup` and `teardown` make sense since they're sort of part of the "public" API (every profiler is expected to have a way to set it up and to tear it down) but `ensure_running` is very implementation-specific. (Maybe your profiler doesn't use a thread or doesn't need to be babysat via `ensure_running` for any reason.)